Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1798,8 +1798,18 @@ protected function convertParameterReflectionToArray(ParameterReflection $parame
 
         $parameterType = $parameter->getType();
         if ($parameterType !== null) {
-            // TODO: This needs to handle ReflectionUnionType
-            $parameterType = ($parameterType instanceof \ReflectionNamedType) ? $parameterType->getName() : $parameterType->__toString();
+            if ($parameterType instanceof \ReflectionUnionType || $parameterType instanceof \ReflectionIntersectionType) {
+                // ReflectionUnionType as of PHP 8, ReflectionIntersectionType as of PHP 8.1
+                $parameterType = implode('|', array_map(
```

Comment:
What do we do further with the reflection information where union types could blow things up?

Reply:
Nothing that would break now - the resulting string is the same as with `__toString()`.

**Except I made a mistake** and the intersection must be imploded using `&`!